### PR TITLE
fix: Logging to stdout in Cloud Run creates a JSON object as "message"

### PR DESCRIPTION
**Pull request by losalex**
This is a fix to provide a configurable way to disable "message" wrapper added for structured log data when LogSync is used for logged entries

Fixes #<[1301](https://github.com/googleapis/nodejs-logging/issues/1301)> 🦕

### CODE REVIEW

File: src/entry.ts [hunk 1]
```diff
@@ -243,7 +245,7 @@ class Entry {
    * Serialize an entry to a standard format for any transports, e.g. agents.
    * Read more: https://cloud.google.com/logging/docs/structured-logging
    */
-  toStructuredJSON(projectId = '') {
+  toStructuredJSON(projectId = '', useMessageField = true) {
```
Comment (daniel-sanche) [inline src/entry.ts:248]: I'm a bit confused by this. Will end users be setting `useMessageField` manually? I don't think other languages work this way

File: src/entry.ts [hunk 2]
```diff
@@ -273,9 +275,22 @@ class Entry {
...
         : undefined;
     // Format log payload.
-    entry.message =
+    let data =
       meta.textPayload || meta.jsonPayload || meta.protoPayload || undefined;
-    entry.message = this.data || entry.message;
+    if (useMessageField) {
+      entry.message = this.data || data;
+    } else {
+      data = this.data || data;
```
Comment (daniel-sanche) [inline src/entry.ts:283]: nit: this block could probably use some comments

File: src/entry.ts (hunk 1 above)
Comment (losalex) [reply src/entry.ts:248]: `useMessageField` will be set through options for those users who does not want to have extra "message" field in `jsonPayload` while running in Cloud Run/Functions and using `stdout` logging. Indeed, other languages dont have this bug - I am fixing this just to make sure not to break existing users which use `LogSync` and already get used to the extra "message" field in `jsonPayload`.

Comment (losalex) [reply src/entry.ts:248]: Yep - for next major release we will turn this flag to be false by default

File: src/entry.ts (hunk 2 above)
Comment (losalex) [reply src/entry.ts]: Added

File: src/entry.ts [hunk 3]
```diff
@@ -273,9 +275,22 @@ class Entry {
...
+      if (data !== undefined && data !== null) {
+        if (this.isObject(data)) {
+          entry = extend(true, {}, entry, data);
+        } else if (typeof data === 'string') {
+          entry.message = data;
+        } else {
+          entry.message = JSON.stringify(data);
```
[the remark this answers is not in the document]
Comment (losalex) [reply src/entry.ts:297]: Yep, mostly - I will definitely add non-text simple test (e.g. numeric)

**Review by daniel-sanche — APPROVED** [review]

**Pull request merged**